shared/domain/src/test/java/com/ivy/domain/usecase/balance/BalanceBuilderPropertyTest.kt

```diff
@@ -0,0 +1,149 @@
+package com.ivy.domain.usecase.balance
+
+import com.ivy.base.TestDispatchersProvider
+import com.ivy.data.model.getFromAccount
+import com.ivy.data.model.getToAccount
+import com.ivy.data.model.primitive.AssetCode
+import com.ivy.data.model.testing.ModelFixtures
+import com.ivy.data.model.testing.transaction
+import com.ivy.domain.statSummary
+import com.ivy.domain.usecase.BalanceBuilder
+import com.ivy.domain.usecase.account.AccountStatsUseCase
+import io.kotest.matchers.shouldBe
+import io.kotest.property.Arb
+import io.kotest.property.arbitrary.filter
+import io.kotest.property.arbitrary.list
+import io.kotest.property.checkAll
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import org.junit.Before
+import org.junit.Test
+
+class BalanceBuilderPropertyTest {
```

Comment:
Add test for:

**property - order of transactions doesn't matter**
- generate an arbitrary list of transactions (original)
- shuffle them (shuffled)
- veify that the balance of `original` and `shuffled` is the same

Again forget the BalanceBuilder, let's make property-tests on the AccountBalanceUseCase

**property - zero balance when each trn is reverted**
- generate an arb list of transactions (trns)
- for each transaction create a 2nd "reverse transaction" (e.g. Income -> Expense, Transfer -> just swap from-to)
- expect that when you calculate balance of both list that it's zero
- verify that balance(trns) == -balance(reverse)
- balance(trna + reverse) == 0